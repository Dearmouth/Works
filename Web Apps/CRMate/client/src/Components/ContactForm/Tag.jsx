/**
 *
 * @param {} param: Presentational Tag Element
 */

export const Tag = ({ name, index, handleRemoveTag }) => {


  return (
    <li className="tag-container">
      <div className="tag-container__name">
        {name}
      </div>
 
      <div className="tag-container__name-popout">
        {name}
      </div>
 
      <div
        onClick={() => handleRemoveTag(index)}
        className="tag-container__cross"
      >
        X
      </div>
    </li>
  );
};
