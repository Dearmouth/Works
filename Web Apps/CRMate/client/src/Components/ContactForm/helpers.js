const numericRegex = /^[0-9]*$/;
const emailRegex = /\w+@\w+\.\w{2,10}/;

export const validateEmail = (email) => email.match(emailRegex);
export const validatePhone = (phone) => phone.match(numericRegex);
