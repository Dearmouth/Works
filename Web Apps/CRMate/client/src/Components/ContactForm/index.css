.form-container {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  
}

.tag-container {
  border-radius: 45px;
  background: white;
  border: 1px solid black;
  display: flex;
  padding: 0.5rem;
  flex: 1 1;
  flex-wrap: none;
  margin: 0.4rem;
  font-size: 15px;
  max-width: 50%;
  gap: 5px;
  min-width: 50px;
  justify-content: space-around;
}

.tag-container__name {
  white-space: no-wrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 80%;
  cursor: pointer;
}

.tag-container__name-popout {
  font-size: 0;
  z-index: -6;
}

.tag-container__name:active + .tag-container__name-popout {
  position: absolute;
  top:100px;
  right: 100px;
  transition: 0.1s;
  max-width: fit-content;
  border-radius: 45px;
  background-color: #c5efa4;
  border: 1px solid black;
  margin: 0 1em 0 1em;
  cursor: pointer;
  padding: 0.5rem;
  overflow: visible;
  width: fit-content;
  font-size: 140%;
  font-weight: 900;
  z-index: 6;
}


.body-wrapper-popup .form-container {
  height: calc(98vh - 350px);
}
.add-tag {
  cursor: pointer;
  font-weight: 900;
}

.tag-container__cross {
  color: red;
  font-weight: bold;
}

.tag-container__cross:hover {
  cursor: pointer;
}

.cross-btn:hover {
  cursor: pointer;
  opacity: 0.8;
  transition: opacity 0.2s ease-in-out;
}

@media only screen and (min-width: 1200px) {
  .body-wrapper {
    background: #e3e3e3;
    padding: 5rem 5rem 0 5rem;
    justify-content: center;
    position: relative;
    left: 331px;
    width: calc(100vw - 331px - 10rem);
    height: calc(100vh - 5rem);
    font-family: "Sanchez One";
  }

  .body-wrapper-popup {
    background: #fff;
    padding: 4rem;
    border-radius: 30px;
    border: 5px solid #000;
    justify-content: center;
    position: fixed;
    font-size: 18px;
    top: 151px;
    bottom: 100px;
    left: calc(50vw - 30vw + 104px);
    width: 60vw;
    z-index: 5;
    font-family: "Sanchez One";
  }
}

@media only screen and (max-width: 1200px) {
  .contact-form-wrapper {
    background: #c4e3ffaf;
    position: absolute;
    width: 100vw;
    height: 100vh;
    z-index: -10;
  }

  .tag-container__name:active + .tag-container__name-popout {
    position: absolute;
    top: 25px;
    left: 45vw;
    font-size: 2.5vw;
  }
  
  .body-wrapper {
    padding: 1.2rem;
    border-color: black;
    border-width: 30px;
    justify-content: center;
    position: fixed;
    font-size: 2.5vw;
    left: calc(5.5vw - 19px);
    top: 140px;
    width: 90vw;
    height: calc(98vh - 100px);
    z-index: -1;
    border: 1px solid #aeaeae;
    background: linear-gradient(#fafafa80, #93c9fc83);
    backdrop-filter: blur(15px);
    border-radius: 25px;
    font-family: "Sanchez One";
    overflow: scroll;
    box-shadow: 1px 1px 10px 4px rgb(0 0 0 / 25%);
  }

  .body-wrapper-popup {
    padding: 1rem;
    border-color: black;
    border-width: 30px;
    justify-content: center;
    position: fixed;
    font-size: 2.5vw;
    left: calc(10vw - 20px);
    top: 140px;
    width: 80vw;
    height: calc(98vh - 250px);
    z-index: 5;
    border: 5px solid #000;
    background: #ffffff50;
    backdrop-filter: blur(15px);
    border-radius: 30px;
    font-family: "Sanchez One";
  }

  .body-wrapper-popup .form-container {
    height: calc(98vh - 300px);
    overflow: hidden;
  }

  .form-cancel {
    border: 3px solid red;
  }

  .form-submit {
    border: 3px solid green;
    margin-left: 2%;
  }
}

.form-label {
  justify-content: space-between;
  font-weight: bold;
  padding: 0.5rem 0;
  flex: 0.5;
}

.form-container .incomplete {
  border: none;
  outline: none;
  box-shadow: 0 0 0 2px red;
}

.form-container textarea {
  border: 2px solid black;
  padding: 0.5rem;
  flex: 2;
}

.btn-with-shadow {
  border-radius: 40px;
  font-weight: bold;
  background: white;
  margin: 0 0 2% 0;
  font-size: 20px;
  padding: 5px 30px 5px 30px;
  box-shadow: 0 4px 4px rgb(0 0 0 / 25%);
  cursor: pointer;
}

.form-cancel {
  border: 1px solid red;
}

.form-delete {
  background: #fcc0c0;
  border: 1px solid red;
}

.form-submit {
  border: 1px solid green;
  margin-left: 2%;
}

.tag-wrapper__input-text {
  border: none;
  border-bottom: 3px solid black;
  outline: none;
  margin-right: 2px;
  display: inline;
}

.tag-wrapper {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}

.tag-wrapper__tags {
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  max-width: 70vw;
  padding: 0;
  margin: 8px 0 0 0;
}

.tag-wrapper__input {
  flex: 1;
  min-width: 20%;
  padding: 0.5rem;
  border: none !important;
  border-bottom: 1px solid black !important;
  outline: none;
}

#icon-check {
  position: absolute;
  right: 25px;
  top: 10px;
}

.form-container__fields__field {
  display: flex;
  justify-content: space-between;
  gap: 10px;
  padding: 0.1rem;
  line-height: 1rem;
}

.form-container__fields__field input {
  padding: 0.5rem;
  border: 2px solid black;
  align-items: center;
  flex: 2;
}

.form-container__fields {
  height: 80%;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  overflow-y: auto;
}
