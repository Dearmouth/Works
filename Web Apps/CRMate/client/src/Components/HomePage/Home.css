/* general */
#App {
  background: #ffffff;
  width: 100vw;
  height: 100vh;
}

/* area tag in #container */
#container > img {
  grid-area: image;
  max-height: 90%;
}

#container > h3 {
  grid-area: text;
}

/* Desktop */
@media (min-width: 1200px) {
  #home-logo {
    position: fixed;
    width: 90px;
    height: 120px;
    left: 20px;
    top: 30px;
  }

  #home-logo-text {
    position: fixed;
    left: 120px;
    font-family: "Sanchez One";
    font-size: 60px;
    line-height: 77px;
    color: #000000;
  }

  #green-bg {
    position: absolute;
    width: 100vw;
    height: 50vh;
    top: 133px;
    background: #77c77a;
    font-family: "Sanchez One";
    line-height: 35px;
  }
  #homeAnimation {
    position: absolute;
    width: 40vw;
    height: 35vh;
    left: 5%;
    top: 54px;
  }

  #welcome {
    position: absolute;
    left: 52vw;
    top: 10%;
    font-style: italic;
    font-size: 12vh;
    text-align: left;
    color: white;
  }

  #welcome-sub {
    position: absolute;
    left: 52vw;
    top: 45%;
    font-style: italic;
    font-size: 3vh;
    text-align: left;
    color: white;
  }

  #intro {
    position: absolute;
    left: 52vw;
    top: 60%;
    font-size: 3vh;
    font-weight: 200;
    text-align: left;
    color: #000000;
  }

  #container {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
  }
  .buttom-wrapper {
    display: flex;
    position: fixed;
    width: 100%;
    height: 31vh;
    bottom: 20px;
    background: #e8e8e8;
    opacity: 0.8;
    flex-direction: row;
  }

  .home-buttons {
    position: absolute;
    top: 6vh;
    right: 5vw;
    display: flex;
    grid-column-gap: 10px;
    grid-column-gap: 10px;
    column-gap: 10px;
    width: 40vw;
    height: 20vh;
    flex-direction: row;
  }

  .home-button {
    height: 5vh;
    width: 20vw;
    padding: 0 4vw 0 4vw;
    background: #c5efa4;
    font-family: "Sanchez One";
    font-size: 2.5vh;
    color: #000000;
    border-style: hidden;
    border-radius: 36px;
    cursor: pointer;
    border: 1px solid black;
  }
  .home-button:hover {
    background: white;
  }
}

/* mobile */
@media (max-width: 1200px) {
  #home-logo {
    position: absolute;
    width: 30vw;
    height: 20vh;
    top: 2vh;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  }

  #home-logo-text {
    position: absolute;
    bottom: 80vh;
    right: 5vw;
    font-family: "Sanchez One";
    font-style: italic;
    font-weight: normal;
    font-size: 8vh;
    overflow: hidden;
    color: #000000;
  }

  .home-buttons {
    position: absolute;
    top: 13vh;
    right: 5vw;
    display: flex;
    grid-column-gap: 10px;
    column-gap: 10px;
    width: 60vw;
    height: 20vh;
    flex-direction: row;
  }

  .home-button {
    height: 5vh;
    width: 30vw;
    padding: 0 4vw 0 4vw;
    background: #c5efa4;
    font-family: "Sanchez One";
    font-size: 2vh;
    color: #000000;
    border-style: hidden;
    border-radius: 36px;
    cursor: pointer;
  }

  #logout-text {
    display: none;
  }

  #green-bg {
    font-family: "Sanchez One";
    font-style: italic;
    font-weight: normal;
    position: absolute;
    display: flex;
    width: 100vw;
    height: 52vh;
    top: 20vh;
    background: #77c77a;
    flex-direction: column;
    align-items: center;
    grid-gap: 10px;
    grid-gap: 0vh;
    gap: 0.2vh;
    line-height: 0px;
    justify-items: center;
    padding: 0 0px 30px 0px;
    overflow: hidden;
    text-align: center;
  }
  #homeAnimation {
    width: 80vw;
    height: 30vh;
    border-radius: 36px;
    padding: 20px;
  }

  #welcome {
    font-size: 8vh;
    letter-spacing: 5px;
    color: #ffffff;
  }

  #welcome-sub {
    font-size: 2.5vh;
    color: #ffffff;
    line-height: 15px;
    
  }

  #intro {
    font-size: 2vh;
    color: #000000;
    line-height: 15px;

  }

  .buttom-wrapper {
    display: flex;
    position: fixed;
    width: 100%;
    height: 21vh;
    bottom: 20px;
    background: #e8e8e8;
    opacity: 0.8;
    flex-direction: column;
  }

  #container {
    width: 100vw;
    height: 7vh;
    font-size: 1.5vh;
    display: grid;
    grid-template-rows: 100%;
    grid-template-areas: "image text";
  }
}
