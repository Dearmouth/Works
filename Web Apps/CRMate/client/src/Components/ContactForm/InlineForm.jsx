import React from "react";
import { Tag } from "./Tag";
/**
 * @description An inline form element that facilitates addition of multiple elements
 * @param {} name: Name of the nested element in the formValue object
 * @param {} values: Initial values
 * @param {*} setValues: Setter function that allows changes to be saved.
 */

export const InlineForm = ({
  maxLength,
  validate = () => true,
  name,
  formValues,
  setFormValues,
  toaster,
}) => {
  let handleRemoveBlur = () => {
    document.getElementById(`${name}-input`) &&
      document.getElementById(`${name}-input`).classList.remove("incomplete");
  };

  let handleRemove = (index) => {
    setFormValues({
      ...formValues,
      [name]: formValues[name].filter((f, i) => i !== index),
    });
  };

  let handleAddTag = (e) => {
    handleRemoveBlur();
    if (e.key !== "Enter" || e.target.value === "") {
      return;
    }

    if (!validate(e.target.value)) {
      // then don't submit
      let currInput = document.getElementById(`${name}-input`);
      currInput.classList.add("incomplete");
      toaster("Invalid input!");

      return;
    }

    let currTags = formValues[name];
    let currTag = document.getElementById(`${name}-input`).value;
    if (!currTag) return;
    currTags ? currTags.push(currTag) : (currTags = [currTag]);
    setFormValues({ ...formValues, [name]: currTags });
    document.getElementById(`${name}-input`).value = "";
  };

  return (
    <div className="tag-wrapper">
      <ul className="tag-wrapper__tags">
        {formValues[name] &&
          formValues[name].map((t, i) => (
            <Tag key={i} name={t} index={i} handleRemoveTag={handleRemove} />
          ))}
      </ul>

      <input
        type="text"
        maxLength={maxLength}
        id={`${name}-input`}
        className="tag-wrapper__input"
        name={name}
        onKeyUp={handleAddTag}
        onChange={handleRemoveBlur}
        placeholder="Add"
      />
    </div>
  );
};
