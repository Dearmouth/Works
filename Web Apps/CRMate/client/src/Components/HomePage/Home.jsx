import logo from "../../Assets/logo.svg";
import homeAnimation from "../../Assets/homeAnimation.gif";
import exportFiles from "../../Assets/export.svg";
import searchPeople from "../../Assets/searchPeople.svg";
import addPeople from "../../Assets/addPeople.svg";
import { useHistory } from "react-router-dom";
import "./Home.css";
import { useMediaQuery } from "react-responsive";
import util from "../../Tools/utils";
import LogoutButton from "../Logout/LogoutButton";
function Home() {
  const isDesktopOrLaptop = useMediaQuery({
    query: "(min-width: 1200px)",
  });
  let history = useHistory();

  return (
    <div>
      <header>
        <img src={logo} id="home-logo" alt="logo" />
        <h1 id="home-logo-text">CRMate</h1>
        <div className = "home-buttons">
        {util.isAuthenticated() ? (
          <button className = "home-button" name = "dashboard-btn" onClick={() => {history.push('/' + localStorage.getItem("nav-selector")) }}>
            Dashboard
          </button>
        ) : (
          <button
            className = "home-button"  
            style = {{background:"#ECECEC"}}
            onClick={() => history.push("/Register")}
            name = "register-btn"
          >
            Register
          </button>
        )}
        {util.isAuthenticated() ? (
          <LogoutButton />
        ) : (
          <button className = "home-button" name = "login-btn" onClick={() => history.push("/Login")}>
            Login
          </button>
        )}
        </div>
      </header>

      <div id="green-bg">
        <h1 id="welcome">Welcome!</h1>
        <div id="welcome-sub">
          {isDesktopOrLaptop ? (
            <div>CRMate, a personal CRM system</div>
          ) : (
            <div> A personal CRM system</div>
          )}
        </div>
        <img src={homeAnimation} id="homeAnimation" alt=" Not found" />
        <div id="intro">A powerful personal relationship management tool</div>
      </div>

      <div className="buttom-wrapper">
        <div id="container" style={{ background: "#FAFAFA" }}>
          <img src={addPeople} alt="addPeople" />
          <h3>Easily save details of anyone you encountered.</h3>
        </div>

        <div id="container" style={{ background: "#ECECEC" }}>
          <img src={searchPeople} alt="searchPeople" />
          <h3>Search for people whenever you need. </h3>
        </div>

        <div id="container" style={{ background: "#F4F4F4" }}>
          <img src={exportFiles} alt="exportFiles" />
          <h3>Transfer Data by downloading it in a useful format </h3>
        </div>
      </div>
    </div>
  );
}

export default Home;
