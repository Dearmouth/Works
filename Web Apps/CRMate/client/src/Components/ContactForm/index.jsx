import React, { useEffect, useState } from "react";
import "./index.css";
import apiService from "../../Tools/apiIndex";
import { InlineForm } from "./InlineForm";
import { Loader } from "../Presentational/Loader";
import { getUserId } from "../../Tools/utils";
import { ENTER_KEY } from "../../Tools/utils";
import { useHistory } from "react-router-dom";
import NavBar from "../NavBar/NavBar";
import { validatePhone, validateEmail } from "./helpers";
import { toast } from "react-toastify";

function ContactForm({ cancelHandler, comingInfo, editing = false }) {
  let history = useHistory();
  let [loading, setLoading] = useState(false);
  let [formValues, setFormValues] = useState(false);
  const toastID = React.useRef(null);
  const createToast = (msg) => {
    if (!toast.isActive(toastID.current)) {
      toastID.current = toast.warning(msg);
    }
  };

  editing
    ? localStorage.setItem("nav-selector", "People")
    : localStorage.setItem("nav-selector", "AddContact");

  useEffect(() => {
    // sets the initial values based on whether this is used for editing or adding
    // editing => accept from route params.
    // adding => initialise to empty fields.
    let init = editing
      ? comingInfo
      : {
          firstName: "",
          lastName: "",
          email: [],
          phone: [],
          tags: [],
          notes: "",
        };
    setFormValues(init);
  }, [comingInfo, editing]);

  // Whenever an input changes
  const handleChange = (e) => {
    // remove blur on any elements
    document.getElementById(`first-name`).classList.remove("incomplete");
    setFormValues({ ...formValues, [e.target.name]: e.target.value });
  };

  const onKeyDown = (keyEvent) => {
    if ((keyEvent.charCode || keyEvent.keyCode) === ENTER_KEY) {
      if (keyEvent.target.type !== "textarea") {
        keyEvent.preventDefault();
      }
    }
  };

  // Edit or Add Contact
  const submit = () => {
    setLoading(true);
    localStorage.setItem("nav-selector", "People");
    if (editing) {
      apiService
        .updateContact(getUserId(), comingInfo._id, formValues)
        .then((res) => {
          setLoading(false);
          cancelHandler();
        });
      window.location.reload();
    } else {
      apiService
        .makeNewContact(getUserId(), formValues)
        .then((res) => {
          setLoading(false);
          document.getElementById("contact-form").reset();
          history.push("/People");
        })
        .catch((e) => {
          setLoading(false);
        });
    }
  };

  // performs basic validation and ensures all changes have been saved to multi-input fields
  // ie. the user has pressed enter and successfully added the new element to the
  // tag, email, phoneNumber list.
  const handleSubmitClick = (e) => {
    e.preventDefault();
    // Don't submit the form if enter is pressed
    // Enter is used in this form to create new tags
    if (e.target.keycode === ENTER_KEY) {
      return false;
    }
    if (!formValues.firstName) {
      // If the user hasn't included a first name, don't submit the form as the entry is invalid
      createToast("Please enter a first name!");
      document.getElementById("first-name").classList.add("incomplete");
      document.getElementById("first-name").focus();
      return;
    } else {
      // check anywhere where the user hasn't pressed enter
      let emailInput = document.getElementById("email-input");
      let phoneInput = document.getElementById("phone-input");
      let tagInput = document.getElementById("tags-input");
      // now warn the user on the display that their inputs have not been saved
      if (emailInput.value || phoneInput.value || tagInput.value) {
        if (tagInput.value) {
          tagInput.classList.add("incomplete");
          createToast("Please ensure you have pressed enter to save this tag.");
        }
        if (phoneInput.value) {
          phoneInput.classList.add("incomplete");
          createToast(
            "Please ensure you have pressed enter to save this phone number."
          );
        }
        if (emailInput.value) {
          emailInput.classList.add("incomplete");
          createToast(
            "Please ensure you have pressed enter to save this email."
          );
          emailInput.focus();
        }
        return;
      }
      submit();
    }
  };

  if (loading) {
    <div>
      {" "}
      <NavBar />
      return <Loader />
    </div>;
  }

  return (
    <div>
      <div className="contact-form-wrapper" />
      {editing ? <></> : <NavBar />}
      <div className={editing ? "body-wrapper-popup" : "body-wrapper"}>
        <form
          id="contact-form"
          onSubmit={handleSubmitClick}
          onKeyDown={onKeyDown}
        >
          <div className="form-container">
            <h1
              style={{
                fontSize: "calc(3vw + 2vh)",
                textDecoration: "underline",
                lineHeight: "0px",
              }}
            >
              {editing ? "Edit Contact" : "Add Contact"}
            </h1>
            <h2>
              {!editing && "Add people to your CRM to never forget them."}
            </h2>

            <div className={"form-container__fields"}>
              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="firstName">
                  First name
                </label>
                <input
                  className="form-input form-input-firstname"
                  id="first-name"
                  value={formValues.firstName || ""}
                  onChange={handleChange}
                  name="firstName"
                ></input>
              </div>

              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="lastName">
                  Last name
                </label>

                <input
                  className="form-input"
                  onChange={handleChange}
                  name="lastName"
                  value={formValues.lastName || ""}
                ></input>
              </div>

              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="email">
                  Email
                </label>
                <InlineForm
                  validate={validateEmail}
                  value={formValues.email || ""}
                  name="email"
                  formValues={formValues}
                  setFormValues={setFormValues}
                  toaster={createToast}
                />
              </div>

              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="phone">
                  Phone
                </label>

                <InlineForm
                  maxLength={30}
                  validate={validatePhone}
                  value={formValues.phone || ""}
                  name="phone"
                  formValues={formValues}
                  setFormValues={setFormValues}
                  toaster={createToast}
                />
              </div>

              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="notes">
                  Notes
                </label>
                <textarea
                  value={formValues.notes || ""}
                  onChange={handleChange}
                  name="notes"
                ></textarea>
              </div>

              <div className="form-container__fields__field">
                <label className="form-label" htmlFor="tags">
                  Tags
                </label>
                <InlineForm
                  value={formValues.tags || ""}
                  name="tags"
                  formValues={formValues}
                  setFormValues={setFormValues}
                />
              </div>
            </div>
            <div>
              <div className="btn-wrapper btn-wrapper-right">
                <button
                  className="btn-with-shadow  form-cancel"
                  onClick={(e) => {
                    e.preventDefault();
                    cancelHandler ? cancelHandler() : history.goBack();
                  }}
                >
                  Cancel
                </button>
                <button
                  id="form-submit"
                  className="btn-with-shadow  form-submit"
                  type="submit"
                >
                  {editing ? "Save" : "Submit"}
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
}

export default ContactForm;
