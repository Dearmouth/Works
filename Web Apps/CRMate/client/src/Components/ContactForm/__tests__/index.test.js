import React from 'react';
import ContactForm from '../index'; 
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import {createMemoryHistory} from 'history'
import { Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from "enzyme";
import Home from "../../HomePage/Home"

configure({ adapter: new Adapter() });

/* Tests the add contact component */
describe("ContactForm component", () => {
  const history = createMemoryHistory();
  history.push('/people')
  it("renders with no errors", async () => {
    const { container } = render(
      <Router history={history}>
      <ContactForm />
      </Router>
    );
  
    expect(container).toMatchSnapshot();
  });

  it("goes back to the home page when the cancel button is pressed", async () => {
    const { container } = render(
      <Router history={history}>
      <ContactForm />
      <Home />
      </Router>
    );

    let cancelButton = container.getElementsByClassName("form-cancel")[0]
    fireEvent.click(cancelButton);
    expect(screen.getByText(/Search for/i)).toBeInTheDocument();
  })

  it("focuses the first name input if submission is attempted without a first name", async () => {
    window.alert = jest.fn();

    const { container } = render(
      <Router history={history}>
      <ContactForm />
      </Router>
    );

    let firstNameInput = container.getElementsByClassName("form-input-firstname")[0];
    fireEvent.change(firstNameInput, {target: {value: ''}})
    fireEvent.click(container.getElementsByClassName("form-submit")[0])
    expect(firstNameInput).toHaveFocus();
  })



})
  