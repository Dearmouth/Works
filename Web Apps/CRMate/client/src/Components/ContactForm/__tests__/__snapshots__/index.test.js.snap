// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component renders with no errors 1`] = `
<div>
  <div>
    <div
      class="contact-form-wrapper"
    />
    <div>
      <div
        id="header-bg"
      >
        <div
          class="navbar-title"
        >
          <p
            id="small"
          >
            AddContact
          </p>
        </div>
      </div>
      <div
        id="navbar-bg"
      >
        <div
          id="navbar-logo-wrapper"
        >
          <img
            alt="navbar-logo "
            id="navbar-logo"
            src="logo.svg"
          />
          <div
            id="navbar-logo-text"
            style="font-family: Sanchez One; font-weight: 700;"
          >
            CRMate
          </div>
        </div>
        <div
          id="navbar-grid"
        >
          <a
            href="/People"
          >
            <button
              class="navbar-container"
              id=""
            >
              <img
                alt="people"
                src="People.svg"
              />
              <h3>
                 People 
              </h3>
            </button>
          </a>
          <a
            href="/Favorite"
            transition="default-fade"
          >
            <button
              class="navbar-container"
              id=""
            >
              <img
                alt="favorite"
                src="Favorite.svg"
              />
              <h3>
                Favorite
              </h3>
            </button>
          </a>
          <a
            href="/AddContact"
          >
            <button
              class="navbar-container"
              id="add"
            >
              <img
                alt="addPeople"
                src="addPeople.svg"
              />
              <h3>
                Add
              </h3>
            </button>
          </a>
          <a
            href="/Data"
          >
            <button
              class="navbar-container"
              id=""
            >
              <img
                alt="dataMigration"
                src="DataMigration.svg"
              />
              <h3>
                Import/Export
              </h3>
            </button>
          </a>
        </div>
        <button
          id="navbar-logout"
        >
          <a
            href="/"
          />
          <img
            alt="logout"
            src="logout.svg"
          />
          <div
            id="navbar-logout-text"
          >
            Logout
          </div>
        </button>
      </div>
    </div>
    <div
      class="body-wrapper"
    >
      <form
        id="contact-form"
      >
        <div
          class="form-container"
        >
          <h1
            style="font-size: calc(3vw + 2vh); text-decoration: underline; line-height: 0px;"
          >
            Add Contact
          </h1>
          <h2>
            Add people to your CRM to never forget them.
          </h2>
          <div
            class="form-container__fields"
          >
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="firstName"
              >
                First name
              </label>
              <input
                class="form-input form-input-firstname"
                id="first-name"
                name="firstName"
                value=""
              />
            </div>
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="lastName"
              >
                Last name
              </label>
              <input
                class="form-input"
                name="lastName"
                value=""
              />
            </div>
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="email"
              >
                Email
              </label>
              <div
                class="tag-wrapper"
              >
                <ul
                  class="tag-wrapper__tags"
                />
                <input
                  class="tag-wrapper__input"
                  id="email-input"
                  name="email"
                  placeholder="Add"
                  type="text"
                />
              </div>
            </div>
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="phone"
              >
                Phone
              </label>
              <div
                class="tag-wrapper"
              >
                <ul
                  class="tag-wrapper__tags"
                />
                <input
                  class="tag-wrapper__input"
                  id="phone-input"
                  maxlength="30"
                  name="phone"
                  placeholder="Add"
                  type="text"
                />
              </div>
            </div>
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="notes"
              >
                Notes
              </label>
              <textarea
                name="notes"
              />
            </div>
            <div
              class="form-container__fields__field"
            >
              <label
                class="form-label"
                for="tags"
              >
                Tags
              </label>
              <div
                class="tag-wrapper"
              >
                <ul
                  class="tag-wrapper__tags"
                />
                <input
                  class="tag-wrapper__input"
                  id="tags-input"
                  name="tags"
                  placeholder="Add"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              class="btn-wrapper btn-wrapper-right"
            >
              <button
                class="btn-with-shadow  form-cancel"
              >
                Cancel
              </button>
              <button
                class="btn-with-shadow  form-submit"
                id="form-submit"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
